src/aosm/azext_aosm/configuration_models/onboarding_sns_input_config.py

```diff
@@ -19,21 +19,19 @@ class SiteNetworkServicePropertiesConfig:
             "comment": "Azure location to use when creating resources e.g uksouth"
         },
     )
-    operator_resource_group: str = field(
+    operator_resource_group_name: str = field(
         default="",
         metadata={"comment": "The resource group that the operator resources will be hosted in."},
     )
     site_name: str = field(
         default="",
         metadata={"comment": "Name of the site."},
     )
-    nsd_reference: "NSDVReferenceConfig" = (
+    nsd_reference: NSDVReferenceConfig = (
         field(
-            default_factory=lambda: [NSDVReferenceConfig()],
+            default=None,
```

Comment:
I think this is wrong. By removing the list you can now only provide one NSD reference. I just tested it with 2 and it didnt work

Reply:
default value for this object "nsd_reference" is none if not set.